### PR TITLE
Add storage rings, fix item length calculation for containers with volume_multiplier

### PR DESCRIPTION
**Pull request by Salty-Panda**
#### Summary
Content "Add storage rings, fix item length calculation for containers with volume_multiplier"

#### Purpose of change

Wanted to add storage rings
Realized that they were 90cm long after putting katana inside

#### Describe the solution

Added storage rings
Fixed how item length is calculated, now it takes into account **volume_multiplier** on pockets.

#### Describe alternatives you've considered

Not having cool magical stuff

#### Testing

Spawned, tested item length on storage rings (always 2cm), normal container (proper length based on the longest item inside), and dimensional toolbelt (20% of longest item inside, as it should be)

### CODE REVIEW

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 1]
```diff
@@ -405,5 +405,51 @@
...
+  },
+  {
+    "type": "TOOL_ARMOR",
+    "copy-from": "mring_platinum",
+    "id": "storage_ring_greater",
+    "name": { "str": "greater storage ring" },
+    "price": "9000 USD",
+    "longest_side": "2 cm",
+    "price_postapoc": "1750 USD",
+    "description": "A glistening ring holding a small dimensional space inside.  Capable of storing a medium-sized weapon you can draw instantly",
```
Comment (casswedson) [inline data/mods/Magiclysm/items/enchanted_rings.json:440]:
```suggestion
    "description": "A glistening ring holding a small dimensional space inside.  Capable of storing a medium-sized weapon you can draw instantly.",
```

File: data/mods/Magiclysm/items/enchanted_rings.json [hunk 2]
```diff
@@ -405,5 +405,51 @@
...
+  },
+  {
+    "type": "TOOL_ARMOR",
+    "copy-from": "mring_platinum",
+    "id": "storage_ring_lesser",
+    "name": { "str": "lesser storage ring" },
+    "price": "3000 USD",
+    "longest_side": "2 cm",
+    "price_postapoc": "1000 USD",
+    "description": "A glistening ring holding a small dimensional space inside.  Can store only a little but you can draw items instantly",
```
Comment (casswedson) [inline data/mods/Magiclysm/items/enchanted_rings.json:417]:
```suggestion
    "description": "A glistening ring holding a small dimensional space inside.  Can store only a little but you can draw items instantly.",
```

File: src/item_pocket.cpp [hunk 3]
```diff
@@ -504,6 +504,16 @@ units::mass item_pocket::item_weight_modifier() const
     return total_mass;
 }
 
+units::length item_pocket::item_length_modifier() const
+{
+    units::length total_length = 0_mm;
+    for( const item &it : contents ) {
+        total_length = std::max( static_cast<units::length>( it.length() * data->volume_multiplier ),
```
Comment (actual-nh) [inline src/item_pocket.cpp:511]: Should this be multiplying by the cube root of the volume multiplier instead?

Comment (Salty-Panda) [reply src/item_pocket.cpp:511]: ...Right, elementary mistake

**Comment by Maleclypse** [comment]
Please open a new PR for this targeting the experimental branch.  Was closed as a side effect of deleting the 0.F -dev branch

**Comment by Salty-Panda** [comment]
Will rebase/resolve this and my other PRs this weekend the earliest

**Comment by Salty-Panda** [comment]
Rebased

**Pull request merged**